### ISSUE TITLE
Deprecate the use of 'pdo' in Doctrine database adapter configuration in Imbo-2.x

### ISSUE THREAD

**Issue by christeredvartsen** [imbo/imbo#534]
Currently the Doctrine database adapters allows the use of a PDO instance in the configuration through the `pdo` config option:

```php
$adapter = new Imbo\Database\Doctrine([
    'pdo' => new PDO(/* ... */);
]);
```

This causes some inconsistencies with regards to exceptions and error handling. This feature will be completely removed in Imbo-3.x, and should be deprecated in Imbo-2.x.

Instead Imbo should rely on the following configuration (used by Doctrine):

```php
$adapter = new Imbo\Database\Doctrine([
    'dbname' => 'database',
    'user' => 'username',
    'password' => 'password',
    'host' => 'hostname',
    'driver' => 'pdo_mysql',
]);
```

**Comment by christeredvartsen** [comment]
We should consider deprecating the second parameter for the doctrine-based database adapters as well, as that allows for setting a connection parameter directly, which can originate from a connection created with the `pdo` configuration option.

**Comment by christeredvartsen** [comment]
Resolved in #536.
